### PR TITLE
Fix issues highlighted by name831

### PR DESCRIPTION
**Pull request by ShmoogleOsukami**
Fixed a broken guide link.
Removed a dead OSX support thread link
Updated the blog with recent news.

### CODE REVIEW

File: _data/blog.yml [hunk 1]
```diff
@@ -1,21 +1,11 @@
 posts:
-- title: "Leisure Gridcoin client update v4.0.1.0!"
-  link_primary_url: "https://github.com/gridcoin-community/Gridcoin-Research/releases/tag/4.0.1.0"
-  link_primary_text: "v4.0.1.0 release notes"
+- title: "Leisure Gridcoin client update v4.0.4.0!"
+  link_primary_url: "https://github.com/gridcoin-community/Gridcoin-Research/releases/tag/4.0.4.0"
+  link_primary_text: "v4.0.4.0 release notes"
   link_secondary_url: ""
   link_secondary_text: ""
-- title: "MANDATORY Gridcoin client update v4.0.0.0!"
```
Comment (RoboticMind) [inline _data/blog.yml]: Do we just want to remove the posts about the old updates?

Comment (RoboticMind) [reply _data/blog.yml]: I think the mandatory still should probably be kept to let people know what the last mandatory version was

Comment (ShmoogleOsukami) [reply _data/blog.yml]: Mmm I had though it might be best to at least list the V4 series of releases. I'll fix that real quick.

Comment (RoboticMind) [reply _data/blog.yml]: Yeah just doing the version 4 updates would probably be fine

**Pull request merged**